docs/DeveloperGuide.md

```diff
@@ -4,10 +4,154 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Design
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Parser
+API: [Parser.java]({repoURL}src/main/java/seedu/cafectrl/parser/Parser.java)
 
+![Parser Class Diagram](images/class/Parser.png)
+
+*Figure 1: Parser Class Diagram*
+
+Note that `CafeCtrl` only have access to the interface `ParserUtil` although the run-time type object is `Parser`. With this, we are able to decrease coupling between `CafeCtrl` and `Parser`, allowing for easier maintenance. This also ensures the testability as we could provide mock or stub dependencies during testing, we could isolate the behavior of the class and focus on unit testing without external dependencies. 
+
+![Parser Parsing User Input Sequence Diagram](images/sequence/Parser.png)
```

Comment:
Alt should have multiple braches, if there's only one possible branch, should use opt instead